lib/appsignal/transaction.rb

```diff
@@ -374,6 +374,24 @@ def set_error(error)
         cleaned_error_message(error),
         backtrace ? Appsignal::Utils::Data.generate(backtrace) : Appsignal::Extension.data_array_new
       )
+
+      causes = []
+      while error
+        error = error.cause
+        causes << error if error
```

Comment:
Please do!

Reply:
Some handling here would be good. Can be larger number than 5, like 50.
Please also add a warning log that when this happens.

Reply:
It is obvious (to Romans) that the number between 5 and 50 is 10, so that's what I went with.

(It's also, roughly speaking, the amount that would fill your whole screen with error causes)

I also added an `is_root_cause: false` marker to the last error cause when the limit is hit. As currently implemented, the processor will ignore it and not write it to MongoDB, but a future change could let that value through, and the front-end could use it to _not_ show the "Root cause" banner when it is not, in fact, the root cause.

I went with a debug message over a warning, because this feels like something that a terrible library could inflict on customers, and I wouldn't want to flood their logs with messages about it. If the first ten error causes didn't tell you what the issue was, it's unlikely that cause #11 will.

(Contrast this with, say, `set_error` being given a non-Exception, which is logged with `error` level, because it is clearly a coding error on the user's end)